p2p/types/addr_book.go

```diff
@@ -1,9 +1,20 @@
 package types
 
+//go:generate mockgen -source=$GOFILE -destination=./mocks/addrbook_provider_mock.go github.com/pokt-network/pocket/p2p/types AddrBookProvider
+
+import "github.com/pokt-network/pocket/shared/modules"
+
 // AddrBook is a way of representing NetworkPeer sets
 type AddrBook []*NetworkPeer
 
 // AddrBookMap maps p2p addresses to their respective NetworkPeer.
 //
 // Since maps cannot be sorted arbitrarily in Go, to achieve sorting, we need to rely on `addrList` which is a slice of addresses/strings and therefore we can sort it the way we want.
 type AddrBookMap map[string]*NetworkPeer
+
+// AddrBookProvider is an interface that provides AddrBook accessors
+type AddrBookProvider interface {
+	GetStakedAddrBookAtHeight(height uint64) (AddrBook, error)
+	ValidatorMapToAddrBook(validators map[string]modules.Actor) (AddrBook, error)
```

Comment:
Add a TODO to remove all the `Validator` specific terminology from the interface.

Reply:
Not necessary, already handled in #378 

https://github.com/pokt-network/pocket/pull/378/files#diff-d4569455b8c8737b4fc4fd9fd0be440b504ea90e873af6c25ce01524c9957711R18-R19